lib/installation_utils.pm

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package installation_utils;
+
+use base Exporter;
+use Exporter;
+use strict;
+use testapi qw(check_var get_var set_var);
+
+our @EXPORT = qw (
+  is_remote_installation
+);
+
+sub is_remote_installation {
+    my ($self) = @_;
+
+    return (check_var('BACKEND', 'ipmi') && !check_var('AUTOYAST', '1') || get_var('SES5_DEPLOY'));
```

Comment:
This is the reason why it is negated

```perl
!check_var('AUTOYAST', '1')
```

It is the same existing condition from **tests/autoyast/installation.pm**

Reply:
being in the same condition does not automagically couple it from a logical perspective. 
condition about negate auto_yast was added by @alice-suse  with PR a29497af082ed5360a6f1e2d1208e505730aedb5 and reason for that was to "fix autoyast installation failure on 3rd generation ipmi backend for sle11sp4". 
you are creating a function which attempts to answer generic  question "is this installation is remote one" and basing your logic on parameter which was introduced as a fix for certain type of tests ( autoyast tests) for certain version of os (sle11sp4 )

Reply:
@asmorodskyi, why not? This negated autoyast is necessary to answer the question "is remote installation?" (remote installation as customer perspective, as we are on an acceptance test)

We are considering IPMI backend as remote installation, but only if autoyast is not used, because autoyast is a "local" installation. What was done inside the if-statements is for, and only for, a remote installation.

There are 2 big groups of remote installations, explicit ones (VNC=1, SSH=1) and implicit ones (IPMI, s390x, XEN, etc.). They have to perform a remote installation and the ticket is about avoiding duplication of the existing and future code. So, nobody should add another if (backend = ipmi) on a place and then duplicate it on another place as if (VNC=1) if the behavior is for remote installation .

If someone adds some behavior for remote installation, the function **is_remote_installation** should be used, and it should affect all remote installations. If the function is missing some case, this case has to be added. If the case doesn't fit inside a remote installation, then it is wrong to put that behavior mixed with remote-installation stuff in the first place.

Reply:
but we DO have AUTOYAST=1 + BACKEND=ipmi runs. https://openqa.suse.de/tests/1495366#settings. this `if` does not mean  "if AUTOYAST then no remote installation", but "in case remote installation with AUTOYAST don't run this part of code".

Reply:
@asmorodskyi, after taking a look at the job, I don't see you point. I see no remote installation there.

I will talk to @alice-suse so she can explain it to me.